util.scad

```diff
@@ -90,3 +90,22 @@ module mirrored(vector_set, index = 0) {
     children();
   }
 }
+
+// A shape that can be used for push in parts
+module pin(height, radius) {
+  inner_height = height * 0.875;
+  cap_height = (height - inner_height) / 2;
+  difference() {
+    union() {
+      up(-inner_height/2)
+        cylinder(inner_height, radius, radius, $fn=8);
+      up(inner_height/2)
+        cylinder(cap_height, radius, radius/2, $fn=8);
+      up(-inner_height/2-cap_height)
```

Comment:
ah I see, this is a double-sided pin, which you might even print as its own piece and use between two parts with negatives!

Still, doesn't have to be. If it projected out of z=0, you could just do `mirrored([z]) pin(len/2)` to get a double sided pin from a single sided one.